packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -3726,6 +3726,13 @@ function shouldForceFlushFallbacksInDEV() {
     // Never force flush in production. This function should get stripped out.
     return false;
   }
+
+  // `isFlushingAct` is used by ReactTestRenderer version of `act`.
+  if (isFlushingAct != null) {
+    // Flush callbacks at the end.
+    return isFlushingAct;
+  }
+
   // `IsThisRendererActing.current` is used by ReactTestUtils version of `act`.
   if (IsThisRendererActing.current) {
```

Comment:
The core issue for the bug I'm fixing is that `IsThisRendererActing.current` is not reset to false before we check `shouldForceFlushFallbacksInDEV`, meaning that we miss the `isFlushingAct = true` value, and we schedule the fallback [here](https://github.com/facebook/react/blob/a27a0d704712a3d3386e8229f438eb94bf8e7992/packages/react-reconciler/src/ReactFiberWorkLoop.new.js#L976) instead of committing it.

Reply:
@acdlite what was the expected behavior here? Should we be checking for isFlushingAct first?

Is the test that failed correct or is it actually wrong?